### ISSUE TITLE
Fix to bug in DataGrid/DataGrid.php

### ISSUE THREAD

**Issue by jgarifuna** [Sintattica/atk#55]
/vendor/sintattica/atk/src/DataGrid/DataGrid.php

        line: 480
            change: 
                $this->addComponent('index', 'atk.datagrid.atkdgindex'); // original generates error

            to:
                $this->addComponent('index', __NAMESPACE__ . '\\DataGridIndex'); // by Jorge Garifuna

**Comment by Samuel-BF** [comment]
Hi,

Well spotted.

This fix have been included in 9.0 branch (#35) and is awaiting inclusion along with other bugfixes in the 9.1 branch (#44 ).

**Comment by jgarifuna** [comment]
Thanks for your reply. I've successfully migrated over 85 modules from ATK 6.x to ATK 9.1. These are modules that I've created between 2006 and now, so they've touched most sections of the framework. I'd like to acknowledge you and commend you for migrating the ATK framework to PHP 7, Bootstrap and JQuery. I know it was a huge undertaking and I'd like to thank you for your great effort!

**Comment by jgarifuna** [comment]
Hi Sintattica,

As you may already know, I contributed a few How Tos
<http://atk-framework.com/wiki/ATK_Howtos> in the legacy ATK framework. I
recently migrated over 85 modules from ATK 6.x to ATK 9.1 and I was
wondering if there is a space in the new github section for ATK where I
could contribute documentation about my experience with ATK 9.1 from time
to time.

Kind regards,

Jorge Garifuna
Tel: 213-915-4402
Tel: 314-666-4402

**Comment by dangamba** [comment]
Hello @jgarifuna and @Samuel-BF 
first of all, sorry for the huge delay in getting back to you... :(

As you may already know, we are still using ATK in production for some old yet critical, complex projects. They are based on the 9.1 branch and I'm afraid we don't have the time to test the many improvements that you brought in the past few years.

Nevertheless, it'd be a shame if all these contributions were lost.
So, we created a 10.0 branch to be the base for improvements and advancements over 9.1, which we consider "frozen" to avoid breaking our legacy projects.

We started to merge some PRs into 10.0 but after merging Samuel's #41 we started to have non-trivial merge conflicts. What can be done about that? @Samuel-BF can you resolve the conflicts? Will you need permissions from us?

We also enabled the wiki for ATK's Github repository. @jgarifuna, would it be OK for you to start writing some docs in it?

As the updated README states, please make you PRs from commits on the 10.0 branch.

Let us know what you think. Any feedback is welcome!

Daniele
Sintattica

**Comment by jgarifuna** [comment]
Hi @dangamba 
Thanks for reaching out! I'll be more than happy to contribute. Do you need to give me access to make my contributions?
Also, it'd be nice to migrate to bootstrap 4+, the latest FontAwesome and jQuery 3.3.1+. This will allow ATK to not fall too far behind and at the same time provide the flexibility to leverage UI templates such as [AdminLTE](https://adminlte.io/). Just in case, here are the dependancies of the [AdminLTE template](https://adminlte.io/docs/3.0/dependencies.html).

**Comment by dangamba** [comment]
@jgarifuna I assume that you already have rights to edit the wiki.
If not, please let me know.
+1 for Bootstrap 4, FontAwesome, JQuery and AdminLTE :)

**Comment by Samuel-BF** [comment]
Hi @dangamba,

Nice to hear some news. The 2 projects I led with ATK are dying now (one because the client is disappearing, the second was a prototype which lost the battle against a more "corporate" development), and I have not a lot of free time for now, so I may not have incentive to integrate all the work I did.

Note that I already integrated all these things into [my-atk branch](https://github.com/Samuel-BF/atk/tree/my-atk) in my repository, so maybe a fastest way to integrate everything would be to merge directly my-atk into master.

Also, I encourage you to integrate security fixes to your project asap (but you know, I guess).

**Comment by jgarifuna** [comment]
> @jgarifuna I assume that you already have rights to edit the wiki.
> If not, please let me know.
> +1 for Bootstrap 4, FontAwesome, JQuery and AdminLTE :)

I just checked and I have acccesssssss!!!!! :)

Just started writing the first doc, which I plan to expand as time permits:
[Migrating from ATK 6 to ATK 9+](https://github.com/Sintattica/atk/wiki/Migrating-from-ATK-6-to-ATK-9-plus)

Is there a way you can look into the other enhancements I posted to see if they could be merged into the new 10 branch you created?
https://github.com/Sintattica/atk/issues

**Comment by dangamba** [comment]
Hello guys

@Samuel-BF :
> Note that I already integrated all these things into [my-atk branch](https://github.com/Samuel-BF/atk/tree/my-atk) in my repository, so maybe a fastest way to integrate everything would be to merge directly my-atk into master.

Of course we'd like to have your contributions into the 10.0 branch!
We'd be willing to completely replace our 10.0 branch with your fork.
How could that be done without having to deal with conflicts/PRS?

@jgarifuna :
Concerning your open issues, we think the best would be for you to make PRs from the 10.0 branch but only after we'll have integrated Samuel's contributions (see above). Is this OK for you?
